sdktests/common_tests_events_contexts.go

```diff
@@ -4,6 +4,7 @@ import (
 	"fmt"
 
 	"github.com/launchdarkly/sdk-test-harness/v2/data"
```

Comment:
Most of the changes in this file aren't related to client-side usage (see next comment for one that is), but I revised the logic here for better code sharing. Previously, in each of these test cases I had to construct a matcher that exactly described the whole JSON object we were looking for. I wasn't able to use the `JSONMatchesContext` helper to do that for me, because it only handled the regular schema, not the event schema. But I refactored that helper so now it can do both. As a result, instead of having to describe the whole expected JSON object, we just have to make any expected changes to the context (like removing redacted attributes) and say which redacted attribute names we expect to see, if any.